src/python/bot/fuzzers/libfuzzer.py

```diff
@@ -1589,6 +1592,52 @@ def use_radamsa_mutator_plugin(extra_env):
   return True
 
 
+def use_peach_mutator(extra_env, arguments):
+  """Decide whether or not to use peach mutator, and set up all of the
+  environment variables necessary to do so."""
+  # TODO(mpherman): Include architecture info in job definition and exclude
+  #  i386.
+  if environment.platform() != 'LINUX' or environment.get_value(
+      'MEMORY_TOOL') != 'ASAN':
+    return False
+
+  # Select grammar.
+  grammar = fuzzer_utils.extract_argument(arguments, 'grammar')
```

Comment:
How is the grammar piped to here? I.e. how is the grammar specified for a target such that at this point there is an argument here?

Also, we shouldn't be using `arguments` for these (the usage seems wrong anyway, as we need to specify the argument of the format `-flag=`). That should be libFuzzer-specific arguments. If we want to add additional metadata, we can add a field to StrategyInfo.

Reply:
We add this "grammar=..." in the libfuzzer options. This it how it ends up in the arguments. Is this not how we should do it?

Reply:
IMO we should create a new section in .options for grammar specific stuff, or even an entirely new file like TARGET.grammar or something similar.

We shouldn't combine libFuzzer specific and non-libFuzzer specific options under the one `arguments`, as that creates confusion.

@mbarbella-chromium WDYT?

Reply:
Another reason not to do this is if we don't choose peach strategy, this argument will be passed to the libFuzzer target, which is wrong and can potentially break things.

Again, we can avoid all of this confusion if we pipe it through via a different mechanism.

Reply:
Good point. We definitely don't want to be passing this along if we aren't picking this strategy. Having the separate .grammar file sounds like the cleanest solution to me.

Reply:
I frankly prefer having a new section in the `.options` file instead.
- We already have there support for libFuzzer, AFL, env variables and various sanitizer options.
- That would also keeps things a bit more simple for OSS-Fuzz, as some projects use `.options` files.
- On Chromium side we will re-use the same script for generating `.options` file instead of duplicating it.

Reply:
@mbarbella-chromium @oliverchang Thoughts?

Reply:
A new section seems fine to me as well.

Would this be the first time we're using a new section for something other than arguments? If so, what happens if a header is omitted? I just want to confirm that we won't suddenly have issues if any OSS-Fuzz targets had improperly formed options files, but I suspect the risk would be very low.

Reply:
> Would this be the first time we're using a new section for something other than arguments?

No, only `[libfuzzer]` and `[AFL]` sections' values go into arguments.

`[env]` elements go into `VARIABLE=VALUE` env vars

`[asan]` and other sanitizers sections go into `ASAN_OPTIONS=VARIABLE=VALUE:...` and such

I think these are all different use cases, and having something new isn't harmful.

> If so, what happens if a header is omitted? 

what do you mean by "header"?